airflow/contrib/hooks/gcf_hook.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from time import sleep
+
+import requests
+from googleapiclient.discovery import build
+
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+
+NUM_RETRIES = 5
+
+
+class GCFHook(GoogleCloudBaseHook):
+    """Hook for Google Cloud Functions APIs."""
+    MAX_RETRIES = 1000
+
+    def __init__(self,
+                 api_version,
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None):
+        super(GCFHook, self).__init__(gcp_conn_id, delegate_to)
+        self.api_version = api_version
+
+    def get_conn(self):
+        """Returns a Google Cloud Functions service object."""
+        http_authorized = self._authorize()
+        return build(
+            'cloudfunctions', self.api_version, http=http_authorized,
+            cache_discovery=False)
+
+    def list_functions(self, location):
+        list_response = self.get_conn().projects().locations().functions().list(
+            parent=location).execute(num_retries=NUM_RETRIES)
+        return list_response.get("functions", [])
+
+    def create_new_function(self, location, body):
+        response = self.get_conn().projects().locations().functions().create(
+            location=location,
+            body=body
+        ).execute(num_retries=NUM_RETRIES)
+        operation_name = response["name"]
+        return self._wait_for_operation_to_complete(operation_name)
+
+    def update_function(self, name, body, fields):
+        response = self.get_conn().projects().locations().functions().patch(
+            updateMask=",".join(fields),
+            name=name,
+            body=body
+        ).execute(num_retries=NUM_RETRIES)
+        operation_name = response["name"]
+        return self._wait_for_operation_to_complete(operation_name)
+
+    def upload_function_zip(self, parent, zip_path):
+        response = self.get_conn().projects().locations().functions().generateUploadUrl(
+            parent=parent
+        ).execute(num_retries=NUM_RETRIES)
+        uploadURL = response.get('uploadUrl')
```

Comment:
nit: upload_url for Python variable name

Reply:
Done by @sprzedwojski